yarn-project/simulator/src/avm/avm_tree.test.ts

```diff
@@ -0,0 +1,344 @@
+import { poseidon2Hash } from '@aztec/foundation/crypto';
+import { Fr } from '@aztec/foundation/fields';
+import { type IndexedTreeLeafPreimage } from '@aztec/foundation/trees';
+import { openTmpStore } from '@aztec/kv-store/utils';
+import { NoopTelemetryClient } from '@aztec/telemetry-client/noop';
+import { MerkleTrees } from '@aztec/world-state';
+
+import { AvmEphemeralForest, EphemeralAvmTree, type IndexedInsertionResult } from './avm_tree.js';
+
+let worldStateTrees: MerkleTrees;
+let copyState: MerkleTreeWriteOperations;
+// Up to 64 dummy note hashes
+let noteHashes: Fr[];
+let indexedHashes: Fr[];
+let slots: Fr[];
+let values: Fr[];
+let getSiblingIndex = 21n;
+
+// Helper to check the equality of the insertion results (low witness, insertion path)
+const checkEqualityOfInsertionResults = <Tree_Height extends number>(
+  containerResults: IndexedInsertionResult<IndexedTreeLeafPreimage>[],
+  wsResults: BatchInsertionResult<Tree_Height, 0>[],
+) => {
+  if (containerResults.length !== wsResults.length) {
+    throw new Error('Results length mismatch');
+  }
+  for (let i = 0; i < containerResults.length; i++) {
+    const containerResult = containerResults[i];
+    const wsResult = wsResults[i];
+    expect(containerResult.lowWitness.lowSiblingPath).toEqual(wsResult.lowLeavesWitnessData![0].siblingPath.toFields());
+    expect(containerResult.lowWitness.index).toEqual(wsResult.lowLeavesWitnessData![0].index);
+    expect(containerResult.lowWitness.preimage).toEqual(wsResult.lowLeavesWitnessData![0].leafPreimage);
+    expect(containerResult.insertionPath).toEqual(wsResult.newSubtreeSiblingPath.toFields());
+  }
+};
+
+beforeEach(async () => {
+  const store = openTmpStore(true);
+  worldStateTrees = await MerkleTrees.new(store, new NoopTelemetryClient());
+  copyState = await worldStateTrees.fork();
+
+  noteHashes = Array.from({ length: 64 }, (_, i) => new Fr(i));
+  // We add 128 since the first 128 leaves are already filled in the indexed trees (nullifier, public data)
+  indexedHashes = Array.from({ length: 64 }, (_, i) => new Fr(i + 128));
+
+  slots = Array.from({ length: 64 }, (_, i) => new Fr(i + 128));
+  values = Array.from({ length: 64 }, (_, i) => new Fr(i + 256));
+});
+
+describe('Simple Note Hash Consistency', () => {
+  it('Should do a simple append and check the root starting with empty', async () => {
+    const treeId = MerkleTreeId.NOTE_HASH_TREE;
+    const treeContainer = await AvmEphemeralForest.create(copyState);
+    for (let i = 0; i < noteHashes.length; i++) {
+      await treeContainer.appendNoteHash(noteHashes[i]);
+    }
+    await worldStateTrees.appendLeaves(treeId, noteHashes);
+
+    const wsRoot = (await worldStateTrees.getTreeInfo(treeId, true)).root;
+    const computedRoot = treeContainer.treeMap.get(treeId)!.getRoot();
+    expect(computedRoot.toBuffer()).toEqual(wsRoot);
+
+    const wsSiblingPath = await worldStateTrees.getSiblingPath(treeId, getSiblingIndex, true);
```

Comment:
Since there are not many, would it make sense to compare every single sibling path?